src/renderer.c

```diff
@@ -244,45 +380,46 @@ static unsigned long font_file_read(FT_Stream stream, unsigned long offset, unsi
 }
 
 static void font_file_close(FT_Stream stream) {
-  if (stream && stream->descriptor.pointer) {
+  if (stream && stream->descriptor.pointer)
     SDL_RWclose((SDL_RWops *) stream->descriptor.pointer);
-    stream->descriptor.pointer = NULL;
-  }
+  free(stream);
 }
 
 RenFont* ren_font_load(const char* path, float size, ERenFontAntialiasing antialiasing, ERenFontHinting hinting, unsigned char style) {
-  RenFont *font = NULL;
-  FT_Face face = NULL;
-  
-  SDL_RWops *file = SDL_RWFromFile(path, "rb");
-  if (!file)
-    goto rwops_failure;
+  SDL_RWops *file = NULL; RenFont *font = NULL;
+  FT_Face face = NULL; FT_Stream stream = NULL;
 
+  file = SDL_RWFromFile(path, "rb");
+  if (!file) return NULL;
+  
   int len = strlen(path);
   font = check_alloc(calloc(1, sizeof(RenFont) + len + 1));
-  font->stream.read = font_file_read;
-  font->stream.close = font_file_close;
-  font->stream.descriptor.pointer = file;
-  font->stream.pos = 0;
-  font->stream.size = (unsigned long) SDL_RWsize(file);
-
-  if (FT_Open_Face(library, &(FT_Open_Args){ .flags = FT_OPEN_STREAM, .stream = &font->stream }, 0, &face))
-    goto failure;
-
-  if (FT_Set_Pixel_Sizes(face, 0, (int)(size)))
-    goto failure;
-
   strcpy(font->path, path);
-  font->face = face;
   font->size = size;
+  font->antialiasing = antialiasing;
+  font->hinting = hinting;
+  font->style = style;
 #ifdef LITE_USE_SDL_RENDERER
   font->scale = 1;
 #endif
+
+  stream = check_alloc(calloc(1, sizeof(FT_StreamRec)));
+  if (!stream) goto stream_failure;
+  stream->read = &font_file_read;
+  stream->close = &font_file_close;
+  stream->descriptor.pointer = file;
+  stream->pos = 0;
+  stream->size = (unsigned long) SDL_RWsize(file);
+
+  if (FT_Open_Face(library, &(FT_Open_Args) { .flags = FT_OPEN_STREAM, .stream = stream }, 0, &face) != 0)
+    goto stream_failure;
```

Comment:
According to https://freetype.org/freetype2/docs/reference/ft2-face_creation.html#ft_reference_face, when `FT_OPEN_STREAM` is supplied here, the function will close the stream if any error occurs; so wouldn't this result in a double-close of the stream?

Reply:
Just confirmed this by forcing an error; this results in a double-free.